lib/src/core/bounds.dart

```diff
@@ -20,6 +21,7 @@ class Bounds<T extends num> {
   Bounds<T> extend(CustomPoint<T> point) {
     CustomPoint<T> newMin;
     CustomPoint<T> newMax;
+    // TODO had this any functionality before?
     if (min == null && max == null) {
```

Comment:
With min and max can't be this. This function can be rewrite with no variable declaration like this :
```dart
Bounds<T> extend(CustomPoint<T> point) {
    return Bounds._(
      CustomPoint(
        math.min(point.x, min.x),
        math.min(point.y, min.y),
      ),
      CustomPoint(
        math.max(point.x, max.x),
        math.max(point.y, max.y),
      ),
    );
  }
  ```

Reply:
just copied your version